include/nbl/builtin/hlsl/bxdf/base/cook_torrance_base.hlsl

```diff
@@ -363,11 +385,15 @@ struct SCookTorrance
         spectral_type quo;
         NBL_IF_CONSTEXPR(IsBSDF)
         {
-            spectral_type prefixThroughputWeights = interaction.getPrefixThroughputWeights();
-            spectral_type reflectance = impl::__implicit_promote<spectral_type, typename fresnel_type::vector_type>::__call(_f(hlsl::abs(cache.getVdotH())));
-            const scalar_type scaled_reflectance = hlsl::dot(reflectance, prefixThroughputWeights);
-            quo = hlsl::mix(reflectance / scaled_reflectance,
-                    (hlsl::promote<spectral_type>(1.0) - reflectance) / (scalar_type(1.0) - scaled_reflectance), cache.isTransmission()) * G2_over_G1;
+            NBL_IF_CONSTEXPR(fresnel_type::ReturnsMonochrome)
+                quo = hlsl::promote<spectral_type>(G2_over_G1);
+            else
+            {
+                const scalar_type scaled_reflectance = __getScaledReflectance(_f, interaction, hlsl::abs(cache.getVdotH()));
+                spectral_type reflectance = impl::__implicit_promote<spectral_type, typename fresnel_type::vector_type>::__call(_f(hlsl::abs(cache.getVdotH())));
```

Comment:
you're calling `_f()` twice, make `__getScaledReflectance` return the `_f()` value in a `NBL_REF_ARG(fresnel_type::vector_type) outFresnelValue` side-channel